File: Php/core/DateTimeFactory.php
namespace core{
	use DateTime;
	use DateTimeZone;

	class DateTimeFactory{

		public static function now(DateTimeZone $timezone = null){
			return self::create(null, $timezone);
		}

		public static function create($str, DateTimeZone $timezone = null){
			if($timezone == null){
				return new DateTime($str);
			}
			return new DateTime($str, $timezone);
		}

		public static function toUtc(DateTime $dateTime){
			$utc = clone $dateTime;
			$utc->setTimezone(new DateTimeZone('UTC'));
			return $utc;
		}

	}

}

File: Infrastructure/System/DateTimeFactory.cs
using System;

namespace Infrastructure.System{

	public static class DateTimeFactory{

		public static DateTime Now(TimeZoneInfo timezone = null){
			return Create(DateTime.Now, timezone);
		}

		public static DateTime Create(string str, TimeZoneInfo timezone = null)
		{
		    var dateTime = DateTime.Parse(str);
		    return Create(dateTime, timezone);
		}

        private static DateTime Create(DateTime dateTime, TimeZoneInfo timezone = null)
        {
            return timezone != null ? TimeZoneInfo.ConvertTime(dateTime, timezone) : dateTime;
        }

	    public static DateTime ToUtc(DateTime dateTime){
			return TimeZoneInfo.ConvertTimeToUtc(dateTime);
		}

	}

}